In [231]:
import pandas as pd
import numpy as np
import re

%matplotlib inline

# STEP #1: **OBJECTIVE**
    ## Calculate the average hours per day in 2014 that men 25 to 30 year of age spent their time in primary activities like  Eating-drinking and Sports

# SETEP #2: 
        ## Variables and activity codes that are needed: TUCASEID, TRCODE and TUACTDUR24 present in atusact_2014.dat.
        Also, TUFINLWGT, TEAGE and TESEX present in atussum_2014.dat

In [282]:
activity_1 = pd.read_csv('atusact_2014.dat', usecols=[0, 26, 13])

In [283]:
activity_1 = activity_1.sort_values(by=['TUCASEID'], ascending=True)

In [284]:
activity_1.head()

TUCASEID  TUACTDUR24  TRCODE
0  20140101140007         540   10101
9  20140101140007         605   10101
7  20140101140007          25  181301
6  20140101140007          90  130116
5  20140101140007          20  181301

# Filtering the Eating and Drinking activity

In [285]:
activity_1 = activity_1[(activity_1.TRCODE >= 110000) & (activity_1.TRCODE < 120000)]

In [286]:
activity_1['eat_drinking'] = activity_1['TUACTDUR24']

In [313]:
activity_1.head()

TUCASEID  eat_drinking
0  20140101140007            30
1  20140101140011            45
2  20140101140028            60
3  20140101140168            65
4  20140101140537            30

In [287]:
activity_1 = pd.DataFrame(activity_1, columns=['TUCASEID', 'TRCODE', 'eat_drinking'])

In [288]:
activity_1.head()

TUCASEID  TRCODE  eat_drinking
4   20140101140007  110101            30
23  20140101140011  110101            30
14  20140101140011  110101            15
30  20140101140028  110101            60
61  20140101140168  110101            30

## ORGANIZING DATA WITH MULTIPLE INDEXING

In [289]:
activity_1 = activity_1.set_index(['TUCASEID', 'TRCODE'])

In [291]:
activity_1.head()

eat_drinking
TUCASEID       TRCODE              
20140101140007 110101            30
20140101140011 110101            30
               110101            15
20140101140028 110101            60
20140101140168 110101            30

In [292]:
activity_1 = activity_1.reset_index(1)

In [293]:
activity_1 = pd.DataFrame(activity_1, columns=['eat_drinking'])

In [294]:
activity_1.head()

eat_drinking
TUCASEID                    
20140101140007            30
20140101140011            30
20140101140011            15
20140101140028            60
20140101140168            30

# STEP #3: SUMARIZING Eating and drinking Activity

In [295]:
activity_1 = activity_1.groupby(level='TUCASEID').sum()

In [296]:
activity_1 = activity_1.reset_index(0)

In [298]:
activity_1.head()

TUCASEID  eat_drinking
0  20140101140007            30
1  20140101140011            45
2  20140101140028            60
3  20140101140168            65
4  20140101140537            30

##  Filtering SPORTS ACTIVITY FROM DATA FILE atusact_2014.dat

In [343]:
activity_2 = pd.read_csv('atusact_2014.dat', usecols=[0, 26, 13])

In [344]:
activity_2 = activity_2.sort_values(by=['TUCASEID'], ascending=True)

In [345]:
activity_2.head()

TUCASEID  TUACTDUR24  TRCODE
0  20140101140007         540   10101
9  20140101140007         605   10101
7  20140101140007          25  181301
6  20140101140007          90  130116
5  20140101140007          20  181301

In [346]:
activity_2 = activity_2[(activity_2.TRCODE >= 130000) & (activity_2.TRCODE < 140000)]

In [347]:
activity_2['Sports'] = activity_2['TUACTDUR24']

In [348]:
activity_2 = pd.DataFrame(activity_2, columns=['TUCASEID', 'TRCODE', 'Sports'])

In [349]:
activity_2.head()

TUCASEID  TRCODE  Sports
6    20140101140007  130116      90
100  20140101140610  130131      15
95   20140101140610  130131      15
143  20140101140639  130128      30
334  20140101140852  130134      60

## ORGANIZING DATA WITH MULTIPLE INDEXING

In [350]:
activity_2 = activity_2.set_index(['TUCASEID', 'TRCODE'])

In [351]:
activity_2.head()

Sports
TUCASEID       TRCODE        
20140101140007 130116      90
20140101140610 130131      15
               130131      15
20140101140639 130128      30
20140101140852 130134      60

## SUMARIZING Sports Activity

In [352]:
activity_2 = activity_2.reset_index(1)

In [353]:
activity_2 = pd.DataFrame(activity_2, columns=['Sports'])

In [354]:
activity_2.head()

Sports
TUCASEID              
20140101140007      90
20140101140610      15
20140101140610      15
20140101140639      30
20140101140852      60

In [355]:
activity_2 = activity_2.groupby(level='TUCASEID').sum()

In [356]:
activity_2 = activity_2.reset_index(0)

In [357]:
activity_2.head()

TUCASEID  Sports
0  20140101140007      90
1  20140101140610      30
2  20140101140639      30
3  20140101140852      60
4  20140101140973     120

# STEP #3: MERGING ACTIVITY 1 AND ACTIVITY 2

In [361]:
activities = pd.merge(activity_1, activity_2)

In [385]:
activities.head()

TUCASEID  eat_drinking  Sports
0  20140101140007            30      90
1  20140101140610            30      30
2  20140101140639           117      30
3  20140101140852            60      60
4  20140101140973            90     120

# STEP #4: GETTING THE SUMMARY FILE READY FOR MERGE WEIGHTED VALUES

In [438]:
poption = pd.read_csv('atussum_2014.dat', usecols=[0, 1, 3, 4])

In [439]:
poption['TUCASEID'] = poption['tucaseid']

In [440]:
poption = pd.DataFrame(poption, columns=['TUCASEID', 'TUFINLWGT', 'TEAGE', 'TESEX'])

In [441]:
poption = poption.sort_values(by=['TUCASEID'], ascending=True)

In [443]:
poption.head()

TUCASEID     TUFINLWGT  TEAGE  TESEX
0  20140101140007  4.891316e+06     20      2
1  20140101140011  2.314737e+07     16      2
2  20140101140028  2.094182e+07     65      2
3  20140101140063  2.505899e+06     50      2
4  20140101140168  1.592823e+07     23      2

# STEP #5: MERGIN THE POPTION DATA SET WITH ACTIVITIES

In [444]:
pop_active = pd.merge(poption, activities)

In [445]:
pop_active.head()

TUCASEID     TUFINLWGT  TEAGE  TESEX  eat_drinking  Sports
0  20140101140007  4.891316e+06     20      2            30      90
1  20140101140610  1.158960e+07     28      1            30      30
2  20140101140639  5.332005e+06     85      1           117      30
3  20140101140852  2.182117e+06     67      1            60      60
4  20140101140973  1.268469e+07     62      1            90     120

# STEP #6: FILTERING POPULATION (MEN 25 - 30)

In [446]:
pop_active = pop_active[(pop_active.TEAGE >= 25) & (pop_active.TEAGE < 31) & (pop_active.TESEX == 1)]

In [454]:
pop_active.head()

TUCASEID     TUFINLWGT  TEAGE  TESEX  eat_drinking  Sports
1   20140101140610  1.158960e+07     28      1            30      30
23  20140111131092  1.182679e+07     30      1           105      30
25  20140111131177  6.565650e+06     26      1            30      66
66  20140112130912  5.252287e+06     30      1           125      60
68  20140112130962  6.156939e+06     30      1            50     346

# STEP #7:  CREATING A NEW COLUMN FOR EACH ACTIVITIY WITH WEIGHTED VALUE IN ORDER TO USE THE average hours per day FORMULA

In [455]:
pop_active['eat_drinking_wtd_time'] = pop_active['TUFINLWGT'] * pop_active['eat_drinking']

In [456]:
pop_active['Sports_wtd_time'] = pop_active['TUFINLWGT'] * pop_active['Sports']

In [458]:
pop_active.head()

TUCASEID     TUFINLWGT  TEAGE  TESEX  eat_drinking  Sports  \
1   20140101140610  1.158960e+07     28      1            30      30   
23  20140111131092  1.182679e+07     30      1           105      30   
25  20140111131177  6.565650e+06     26      1            30      66   
66  20140112130912  5.252287e+06     30      1           125      60   
68  20140112130962  6.156939e+06     30      1            50     346   

    eat_drinking_wtd_time  Sports_wtd_time  
1            3.476881e+08     3.476881e+08  
23           1.241813e+09     3.548037e+08  
25           1.969695e+08     4.333329e+08  
66           6.565359e+08     3.151372e+08  
68           3.078470e+08     2.130301e+09

# STEP 8: Getting the formula denominator for both activities

In [459]:
denominator = pop_active.TUFINLWGT.sum(axis=0)

In [460]:
eat_drk_numerator = pop_active.eat_drinking_wtd_time.sum(axis=0)

In [ ]:
eat_drk_ave = eat_drk_numerator / denominator

In [468]:
eat_drk_ave

77.063295239036293

In [469]:
eat_drk_ave_hr_perday = eat_drk_ave / 60

In [470]:
eat_drk_ave_hr_perday

1.2843882539839382

## In 2014, Men 25 to 30 of age spent an average of 1.28 hours per day eating and drinking

In [471]:
sports_numerator = pop_active.Sports_wtd_time.sum(axis=0)

In [451]:
sports_ave = sports_numerator / denominator

In [452]:
sports_ave

112.11763153996382

In [474]:
sports_ave_hr_perday = sports_ave / 60

In [475]:
sports_ave_hr_perday

1.8686271923327304

## In 2014, Men 25 to 30 of age spent an average of 1.86 hours per day doing Some kind of Sports